src/persist/src/nemesis/direct.rs

```diff
@@ -227,22 +219,40 @@ impl Direct {
     }
 
     fn seal(&mut self, req: SealReq) -> Result<Future<()>, Error> {
-        // NB: Once the caller resolves the returned future, it'd probably be
-        // good to wait until the output corresponding to this stream catches up
-        // to the sealed timestamp. Otherwise, we might end up testing the
-        // uninteresting but correct case of no output.
-        let (write, _, _) = self.stream(&req.stream)?;
-        Ok(write.seal(req.ts))
+        let stream = self.stream(&req.stream)?;
+        let seal_ts = req.ts;
+        let seal_res = stream.write.seal(seal_ts);
+        let progress = stream.progress.clone();
+        let (tx, rx) = Future::new();
+        let _ = thread::Builder::new()
+            .name("nemesis-seal".into())
```

Comment:
minor: we maybe also want to add the request number to this name / the seal ts? not sure if thats useful for debugging?

Reply:
I had to debug quite a few deadlocks while getting this all to work and knowing the specific request never ended up being helpful to me in that process, so inclined to skip this one